### PR TITLE
Terrain surface materials list component.

### PR DESCRIPTION
**Pull request by sphrose**
Signed-off-by: sphrose <82213493+sphrose@users.noreply.github.com>

### CODE REVIEW

File: content/docs/user-guide/components/reference/terrain/surface_material_list.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+[Axis Aligned Box Shape](/docs/user-guide/components/reference/shape/axis-aligned-box-shape/), will adopt that material.
+
+
+## Usage ##
+Select the surface type tag using the **SurfaceTag** pull down, then assign a material by using the open button, or by dragging from the AssetBrowser window.
+
+## Provider ##
+
+[Terrain Gem](/docs/user-guide/gems/reference/terrain)
+
```
Comment (FiniteStateGit) [inline content/docs/user-guide/components/reference/terrain/surface_material_list.md:20]:
```suggestion

## Dependencies

The [Axis Aligned Box Shape Component](/docs/user-guide/components/reference/shape/axis-aligned-box-shape) is required for the Terrain Surface Materials List component to operate.

```
Suggest adding a dependencies section to match the format of other component reference pages.

File: content/docs/user-guide/gems/reference/environment/terrain/_index.md [hunk 2]
```diff
@@ -30,4 +30,5 @@ To enable the Terrain Gem, do the following:
 | Component | Description | 
 | - | - |
 | [Terrain Layer Spawner](/docs/user-guide/components/reference/terrain/layer_spawner.md) | Spawns a terrain region contained within configurable bounds, and allows prioritization of overlapping terrain layers. |
```
Comment (FiniteStateGit) [inline content/docs/user-guide/gems/reference/environment/terrain/_index.md:32]:
```suggestion
| [Terrain Layer Spawner](/docs/user-guide/components/reference/terrain/layer_spawner) | Spawns a terrain region contained within configurable bounds, and allows prioritization of overlapping terrain layers. |
```

File: content/docs/user-guide/gems/reference/environment/terrain/_index.md [hunk 3]
```diff
@@ -30,4 +30,5 @@ To enable the Terrain Gem, do the following:
...
-| [Terrain Height Gradient List](/docs/user-guide/components/reference/terrain/terrain_height_gradient_list.md) | Provides terrain height data from a list of gradients. |
\ No newline at end of file
+| [Terrain Height Gradient List](/docs/user-guide/components/reference/terrain/height_gradient_list/) | Provides terrain height data from a list of gradients. |
```
Comment (FiniteStateGit) [inline content/docs/user-guide/gems/reference/environment/terrain/_index.md:33]:
```suggestion
| [Terrain Height Gradient List](/docs/user-guide/components/reference/terrain/height_gradient_list) | Provides terrain height data from a list of gradients. |
```

File: content/docs/user-guide/gems/reference/environment/terrain/_index.md [hunk 4]
```diff
@@ -30,4 +30,5 @@ To enable the Terrain Gem, do the following:
...
+| [Terrain Surface Material List](/docs/user-guide/components/reference/terrain/surface_material_list/) | Defines mappings between a surface type and a render material. |
```
Comment (FiniteStateGit) [inline content/docs/user-guide/gems/reference/environment/terrain/_index.md:34]:
```suggestion
| [Terrain Surface Material List](/docs/user-guide/components/reference/terrain/surface_material_list) | Defines mappings between a surface type and a render material. |
```

File: content/docs/user-guide/components/reference/_index.md [hunk 5]
```diff
@@ -212,7 +212,8 @@ The components below are grouped by type as they appear in the O3DE Editor.
 | Component | Description | 
 | - | - |
 | [Terrain Layer Spawner](/docs/user-guide/components/reference/terrain/layer_spawner/) | Spawns a terrain region contained within configurable bounds, and allows prioritization of overlapping terrain layers. |
```
Comment (FiniteStateGit) [inline content/docs/user-guide/components/reference/_index.md:214]:
```suggestion
| [Terrain Layer Spawner](/docs/user-guide/components/reference/terrain/layer_spawner) | Spawns a terrain region contained within configurable bounds, and allows prioritization of overlapping terrain layers. |
```
Suggesting these url changes to align with edits in the Terrain Collider PR.

File: content/docs/user-guide/components/reference/_index.md [hunk 6]
```diff
@@ -212,7 +212,8 @@ The components below are grouped by type as they appear in the O3DE Editor.
...
-| [Terrain Height Gradient List](/docs/user-guide/components/reference/terrain/terrain_height_gradient_list.md) | Provides terrain height data from a list of gradients. |
+| [Terrain Height Gradient List](./terrain/height_gradient_list/) | Provides terrain height data from a list of gradients. |
```
Comment (FiniteStateGit) [inline content/docs/user-guide/components/reference/_index.md:215]:
```suggestion
| [Terrain Height Gradient List](./terrain/height_gradient_list) | Provides terrain height data from a list of gradients. |
```

File: content/docs/user-guide/components/reference/_index.md [hunk 7]
```diff
@@ -212,7 +212,8 @@ The components below are grouped by type as they appear in the O3DE Editor.
...
+| [Terrain Surface Material List](./terrain/surface_material_list/) | Defines mappings between a surface type and a render material. |
```
Comment (FiniteStateGit) [inline content/docs/user-guide/components/reference/_index.md:216]:
```suggestion
| [Terrain Surface Material List](./terrain/surface_material_list) | Defines mappings between a surface type and a render material. |
```

**Review by FiniteStateGit — CHANGES_REQUESTED** [review]

**Comment by FiniteStateGit** [comment]
I mentioned this in the Terrain Collider PR, but the filenames use underscores, I'm not sure if hyphens should be used.

**Comment by sphrose** [comment]
> 
> 
> I mentioned this in the Terrain Collider PR, but the filenames use underscores, I'm not sure if hyphens should be used.

I've made this change just in case.

**Review by FiniteStateGit — APPROVED** [review]
Looks good, approved.

**Comment by sptramer** [comment]
@sphrose Looks like there are conflicts - can you resolve them and I'll re-review tomorrow?

**Comment by sphrose** [comment]
> 
> 
> @sphrose Looks like there are conflicts - can you resolve them and I'll re-review tomorrow?

Done

**Review by sptramer — APPROVED** [review]

**Comment by sptramer** [comment]
@mbalfour-amzn Please merge after tech review. Thanks!

**Review by mbalfour-amzn — APPROVED** [review]

**Pull request merged**